docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,402 @@
+ml_train_model_destroy(model);
+```
+
+### Layer
+
+Layer is a component that does actual computation while managing internal trainable parameters.
+
+```
+/***************************************
+ * Data(x)  -> Layer(F) -> Data(F(X))  *
+ * Data(dx) <-    ...   <- Data(dF(X)) *
+ ***************************************/
+```
+
+```c
+// Create layer
+ml_train_layer_h layer;
+ml_train_layer_create(&layer, ML_TRAIN_LAYER_TYPE_FC);
+
+// configure layer
+ml_train_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=zeros", NULL);
+
+// after add layer to model,
+// you do not need to destroy layer since ownership is transferred to the model.
+ml_train_model_add_layer(model, layer);
+```
+
+### Optimizer
+
+Optimizer determines how to update model parameters according to loss from prediction.
+Currently, Stochastic gradient descent optimizer and Adam optimizer are supported.
+
+```c
+// Create optimizer
+ml_train_optimizer_h optimizer;
+ml_train_optimizer_create(&optimizer, ML_TRAIN_OPTIMIZER_TYPE_SGD);
+
+// configure optimizer
+ml_train_optimizer_set_property(optimizer, "learning_rate=0.001", NULL);
+
+// after configuring optimizer to model
+// you do not need to destroy optimizer since ownership is transferred to the model.
+ml_train_model_set_optimizer(model, optimizer);
+```
+
+### Dataset
+
+Dataset is in charge of feeding data into the model.
+It can either be created from callback function or created from a file.
+Detail will be discussed in [upcoming](#prepare-model) section.
+
+```c
+// Create dataset
+ml_train_dataset_h dataset;
+ml_train_dataset_create_with_generator(&dataset, generator_train_cb, generator_valid_cb, generator_test_cb);
+
+// configure dataset
+ml_train_dataset_set_property(dataset, "buffer_size=100", NULL);
+
+// after set dataset to model,
+// you do not need to destroy dataset since ownership is transferred to the model.
+ml_train_model_set_dataset(model, dataset);
+```
+
+## Construct Model
+
+To run the model, you need to construct model. Model can be constructed with `ml_train_model_construct`.
+If you have file that describe model, the file can be used to construct initially with `ml_train_model_construct_with_file`.
+Even if you constructed model from a file, you can switch/modify/add each component with set of given apis until you compile with `ml_train_model_compile`.
+
+### Construct Model from the Description File
+
+As of now, only `*.ini` is supported to construct model from a file.
+
+#### Create Model from ini Format
+
+There are special sections refering to `model` and `dataset` respectively [Model] and [Dataset]
+Each ini section maps to a `layer`. Keys and values from each section sets property.
+Invalid Property will be ignored safely, invalid values will cause error. All keys and values are case-insensitive
+
+For example,
+
+```ini
+[Model] # Special section that describes model property
+Type = NeuralNetwork	# Model Type : only NeuralNetwork is supported as of now
```

Comment:
such information `only NeuralNetwork is supported as of now` must not be given here. But rather in docs of the project.